ratatool-sampling/src/main/scala/com/spotify/ratatool/samplers/BigSampler.scala

```diff
@@ -96,15 +113,17 @@ object BigSampler {
   def singleInput(argv: Array[String]): Future[Tap[_]] = {
     val (sc, args) = ContextAndArgs(argv)
     val (opts, _) = ScioContext.parseArguments[PipelineOptions](argv)
-    val samplePct = args("sample").toFloat
 
-    require(samplePct > 0.0F && samplePct <= 1.0F,
-      "Sample percentage should be between (0.0, 1.0]")
-
-    val input = args("input")
-    val output = args("output")
-    val fields = args.list("fields")
-    val seed = args.optional("seed")
+    val (samplePct, input, output, fields, seed) = try {
+      val pct = args("sample").toFloat
+      require(pct > 0.0F && pct <= 1.0F)
+      (pct, args("input"), args("output"), args.list("fields"),
+        args.optional("seed"))
+    } catch {
+      case e: Throwable =>
```

Comment:
do we want to catch only `NonFatal` here instead of everything

http://www.scala-lang.org/api/current/scala/util/control/NonFatal$.html